src/libecalc/presentation/yaml/domain/expression_time_series_flow_rate.py

```diff
@@ -27,8 +38,16 @@ def __init__(
         self._regularity = regularity
         assert isinstance(consumption_rate_type, RateType)
         self._consumption_rate_type = consumption_rate_type
+        self._rate_values = self._get_stream_day_values()
+        self._validate()
 
-    def get_stream_day_values(self) -> list[float | None]:
+    def _validate(self):
+        """Validate that all flow rate values are positive."""
+        for rate in self._rate_values:
+            if rate is not None and rate < 0:
```

Comment:
_get_stream_day_values returns list[float|None] so needs to handle None here as well, i think